dp/generate_climos.py

```diff
@@ -306,36 +296,74 @@ def convert_pr_var_units(input_file, climo_means):
     return climo_means
 
 
-def update_climo_time_meta(filepath):
+def split_on_variables(climo_means_file, var_names):
+    if len(var_names) > 1:
+        return [cdo.select('name={}'.format(var_name), input=climo_means_file)
+                for var_name in var_names]
+    else:
+        return [climo_means_file]
+
+
+def update_metadata_and_time_var(input_file, t_start, t_end, climo_filepath):
     """Updates an existing netCDF file to reflect the fact that it contains climatological means.
-    Namely, update the time variable with climatological times computed according to CF Metadata Convetions,
+
+    Specifically:
+    - add start and end time attributes
+    - update tracking_id attribute
+    - update the frequency attribute
+    - update the time variable with climatological times computed according to CF Metadata Convetions,
     and create a climatology bounds variable with appropriate values.
 
-    :param filepath: (str) filepath to a climatological means output file which needs to have its
-        time variable
+    :param input_file: (CFDataset) input file from which the climatological output file was produced
+    :param t_start: (datetime.datetime) start date of climatological output file
+    :param t_end: (datetime.datetime) end date of climatological output file
+    :param climo_filepath: (str) filepath to a climatological means output file which needs to have its
+        metadata update
 
-    IMPORTANT: THIS MAKES CHANGES TO FILES IN PLACE
+    WARNING: THIS CHANGES FILE `climo_filepath` IN PLACE
```

Comment:
That is directed at the developer. It's in distinction to the CDO operators, which always generate a new file containing the modification. In-place doesn't matter much unless you apply it directly to an input file. In this code, it is always applied to an intermediate results file.